rules/dockerfile_build.bzl

```diff
@@ -0,0 +1,88 @@
+# Copyright 2017 Google Inc. All rights reserved.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Rule for building debootstrap rootfs tarballs."""
+
+def _impl(ctx):
+    # Strip off the '.tar'
+    base_image = ctx.attr.base.label.name.split('.', 1)[0]
+    # docker_build rules always generate an image named 'bazel/$package:$name'.
+    base_image_name = "bazel/%s:%s" % (ctx.attr.base.label.package,
+                                       base_image)
+    dockerfile_path = ctx.file.dockerfile.path
+    context_path = ctx.file.context.path
+
+    # Generate a shell script to run the build.
+    build_contents = """\
+#!/bin/bash
+set -ex
+
+# Load the base image
+docker load -i {base_tar}
+
+# Template out the FROM line.
+sed -i "" "s|FROM.*|FROM {base_name}|g" {dockerfile}
```

Comment:
There is an additional `""` in the sed call.  It is continuing with that change

Reply:
This is weird, I need that "" on OSX, but it appears to break Linux. I'll see if I can come up with  a cross platform way to do this.